lightningd/invoice.c

```diff
@@ -270,7 +270,7 @@ void invoice_try_pay(struct lightningd *ld,
 	tal_add_destructor2(hin, invoice_payload_remove_hin, payload);
 
 	log_debug(ld->log, "Calling hook for invoice '%s'", details->label->s);
-	plugin_hook_call_invoice_payment(ld, payload, payload);
+	plugin_hook_call_invoice_payment(ld->wallet->db, payload, payload);
```

Comment:
Changing this parameter seems strange: the two seem unrelated.  We can get at ld in other ways, but handing it explicitly here is the normal pattern, and plugin_hook isn't a db function.

Reply:
I had to do this to standardize `db_write`. I thought it could have been acceptable since we did not use the `lightningd` struct, only the db.

Reply:
I agree with @rustyrussell here, let's keep it `ld` since hooks are not related to the DB at all.

Reply:
In that case I cannot call the `db_write` hook without modifying the `db` struct.